pickfirst.go

```diff
@@ -75,7 +75,7 @@ func (b *pickfirstBalancer) HandleResolvedAddrs(addrs []resolver.Address, err er
 
 func (b *pickfirstBalancer) HandleSubConnStateChange(sc balancer.SubConn, s connectivity.State) {
 	if grpclog.V(2) {
-		grpclog.Infof("pickfirstBalancer: HandleSubConnStateChange: %p, %v", sc, s)
+		grpclog.Infof("pickfirstBalancer: HandleSubConnStateChange: %+v, %v", sc, s)
```

Comment:
SubConn contains internal-only content that is not to be printed.

What information do you need specifically? (Addresses can be found in another log line before this.)

Reply:
>> What information do you need specifically? (Addresses can be found in another log line before this.)

INFO: 2019/12/18 22:08:00 lbBalancer: server list entry[0]: ipStr:|[2001:4860:8040:49:0:77:109c:8186]|, port:|14861|, load balancer token:|AAAAAAAtAAAAAAAK|
INFO: 2019/12/18 22:08:00 lbBalancer: server list entry[1]: ipStr:|[2001:4860:8040:49:0:77:109c:8187]|, port:|26058|, load balancer token:|AAAAAAAtAAAAAAAK|
INFO: 2019/12/18 22:08:00 lbBalancer: server list entry[2]: ipStr:|[2001:4860:8040:49:0:77:109c:8184]|, port:|14991|, load balancer token:|AAAAAAAtAAAAAAAK|
INFO: 2019/12/18 22:08:00 lbBalancer: server list entry[3]: ipStr:|[2001:4860:8040:49:0:77:109c:818c]|, port:|14838|, load balancer token:|AAAAAAAtAAAAAAAK|
INFO: 2019/12/18 22:08:00 lbBalancer: server list entry[4]: ipStr:|[2001:4860:8040:49:0:77:109c:8185]|, port:|14781|, load balancer token:|AAAAAAAtAAAAAAAK|
INFO: 2019/12/18 22:08:00 lbBalancer: server list entry[5]: ipStr:|[2001:4860:8040:49:0:77:109c:8186]|, port:|14861|, load balancer token:|AAAAAAAtAAAAAAAK|
INFO: 2019/12/18 22:08:00 lbBalancer: server list entry[6]: ipStr:|[2001:4860:8040:49:0:77:109c:8187]|, port:|26058|, load balancer token:|AAAAAAAtAAAAAAAK|
INFO: 2019/12/18 22:08:00 lbBalancer: server list entry[7]: ipStr:|[2001:4860:8040:49:0:77:109c:8184]|, port:|14991|, load balancer token:|AAAAAAAtAAAAAAAK|
INFO: 2019/12/18 22:08:00 lbBalancer: server list entry[8]: ipStr:|[2001:4860:8040:49:0:77:109c:818c]|, port:|14838|, load balancer token:|AAAAAAAtAAAAAAAK|
INFO: 2019/12/18 22:08:00 lbBalancer: server list entry[9]: ipStr:|[2001:4860:8040:49:0:77:109c:8185]|, port:|14781|, load balancer token:|AAAAAAAtAAAAAAAK|
INFO: 2019/12/18 22:08:00 lbBalancer: handle SubConn state change: 0xc0001b9750, CONNECTING
INFO: 2019/12/18 22:08:00 lbBalancer: handle SubConn state change: 0xc0001b9770, CONNECTING
INFO: 2019/12/18 22:08:00 lbBalancer: handle SubConn state change: 0xc0001b9790, CONNECTING
INFO: 2019/12/18 22:08:00 lbBalancer: handle SubConn state change: 0xc0001b97b0, CONNECTING
INFO: 2019/12/18 22:08:00 lbBalancer: handle SubConn state change: 0xc0001b97e0, CONNECTING
INFO: 2019/12/18 22:08:00 lbBalancer: handle SubConn state change: 0xc0001b9750, READY
INFO: 2019/12/18 22:08:00 lbBalancer: handle SubConn state change: 0xc0001b9790, READY
INFO: 2019/12/18 22:08:00 lbBalancer: handle SubConn state change: 0xc0001b97b0, READY
INFO: 2019/12/18 22:08:00 lbBalancer: handle SubConn state change: 0xc0001b9770, READY
INFO: 2019/12/18 22:08:00 lbBalancer: handle SubConn state change: 0xc0001b97e0, READY

These logs don't tell which SubConn is READY or CONNECTING now. It would be useful debug information to add the backend addresses to the state change log line and hence this PR.

Reply:
The balancer you are looking at is grpclb, not pickfirst.

The addresses in the SubConn cannot be simply printed because it can change.
Another way is to print the address for the subconn in the balancer, and the balancer will need to keep the map from each subconn to it's addresses (which it [doesn't](https://github.com/grpc/grpc-go/blob/master/balancer/grpclb/grpclb.go#L216) today).